registry_module_integration_test.go

```diff
@@ -1566,6 +1566,261 @@ func TestRegistryModulesDeleteVersion(t *testing.T) {
+			Name:         "",
+		})
+		assert.Error(t, err)
+		assert.Equal(t, err, ErrRequiredName)
+	})
+
+	t.Run("with invalid org", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByName(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: badIdentifier,
+			RegistryName: PrivateRegistry,
+			Namespace:    "terraform-aws-modules",
+			Name:         registryModuleTest.Name,
+		})
+		assert.EqualError(t, err, ErrInvalidOrg.Error())
+	})
+
+	t.Run("with invalid registry name", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByName(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: orgTest.Name,
+			RegistryName: badIdentifier,
+			Namespace:    "terraform-aws-modules",
+			Name:         registryModuleTest.Name,
+		})
+		assert.Error(t, err)
+	})
+
+	t.Run("with namespace and when registry name is private", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByName(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: orgTest.Name,
+			RegistryName: PrivateRegistry,
+			Namespace:    "terraform-aws-modules",
+			Name:         registryModuleTest.Name,
+		})
+		assert.Error(t, err)
+	})
+}
+
+func TestRegistryModulesDeleteByProvider(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	registryModuleTest, _ := createRegistryModule(t, client, orgTest, PrivateRegistry)
+
+	assert.NotNil(t, orgTest)
+
+	t.Run("with valid parameters", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByProvider(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: orgTest.Name,
+			RegistryName: PrivateRegistry,
+			Namespace:    registryModuleTest.Namespace,
+			Name:         registryModuleTest.Name,
+		}, "provider")
+
+		require.NoError(t, err)
+	})
+
+	t.Run("when the registry module does not exist", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByName(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: orgTest.Name,
+			RegistryName: PrivateRegistry,
+			Namespace:    "terraform-aws-modules",
+			Name:         "",
+		})
+		assert.Error(t, err)
+		assert.Equal(t, err, ErrRequiredName)
+	})
+
+	t.Run("with invalid org", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByName(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: badIdentifier,
+			RegistryName: PrivateRegistry,
+			Namespace:    "terraform-aws-modules",
+			Name:         registryModuleTest.Name,
+		})
+		assert.EqualError(t, err, ErrInvalidOrg.Error())
+	})
+
+	t.Run("with invalid registry name", func(t *testing.T) {
+		err := client.RegistryModules.DeleteByName(ctx, RegistryModuleDeleteByNameOrProvider{
+			Organization: orgTest.Name,
+			RegistryName: badIdentifier,
+			Namespace:    "terraform-aws-modules",
+			Name:         registryModuleTest.Name,
+		})
+		assert.Error(t, err)
+	})
+
+	t.Run("with namespace and when registry name is private", func(t *testing.T) {
```

Comment:
same here

Reply:
@Uk1288 can you restate what you are flagging here?

Reply:
nvm